addons/website_blog/views/website_blog_components.xml

```diff
@@ -21,9 +22,48 @@
                     </a>
                 </li>
             </ul>
+            <!-- Mobile -->
+            <t t-set="navIsOffCanvas" t-value="len(blogs) > 4"/>
+            <div t-if="len(blogs) > 1" t-attf-class="order-2 d-lg-none #{not navIsOffCanvas and 'dropdown flex-grow-1 flex-sm-grow-0 min-w-0'}">
+                <a t-attf-class="btn btn-light #{not navIsOffCanvas and 'dropdown-toggle d-flex align-items-center justify-content-between'}" 
+                   type="button" 
+                   role="button" 
+                   t-attf-title="{{blog.name if blog else 'All Blogs'}}" 
+                   t-att-data-bs-toggle="navIsOffCanvas and 'offcanvas' or 'dropdown'" 
+                   t-att-data-bs-target="navIsOffCanvas and '#o_wblog_offcanvas'" 
+                   t-att-aria-expanded="not navIsOffCanvas and 'false'" 
+                   t-att-aria-controls="navIsOffCanvas and 'o_wblogs_offcanvas'">
+                    <i t-if="navIsOffCanvas" class="fa fa-navicon"/>
+                    <t t-else="">
+                        <span t-if="blog" class="d-inline-block text-truncate" t-out="blog.name"/>
+                        <span t-else="" class="d-inline-block text-truncate">All</span>
+                    </t>
+                </a>
+                <div id="o_wblog_offcanvas" t-att-class="navIsOffCanvas and 'offcanvas offcanvas-end d-lg-none' or 'dropdown-menu dropdown-menu-end'">
+                    <div t-if="navIsOffCanvas" class="offcanvas-header">
+                        <h5 class="offcanvas-title">Blogs</h5>
+                        <button type="button" class="btn-close" data-bs-dismiss="offcanvas" aria-label="Close"/>
+                    </div>                    
+                    <div t-att-class="navIsOffCanvas and 'offcanvas-body'">
+                        <ul t-if="len(blogs) > 1" t-att-class="navIsOffCanvas and 'nav navbar-nav' or 'list-unstyled mb-0'">
+                            <li t-att-class="navIsOffCanvas and 'nav-item'">
+                                <a href="/blog" t-attf-class="#{navIsOffCanvas and 'nav-link' or 'dropdown-item'} #{(not blog) and 'active'}" title="All">All</a>
+                            </li>
+                            <li t-foreach="blogs" t-as="nav_blog" t-att-class="navIsOffCanvas and 'nav-item'">
+                                <a t-attf-href="/blog/#{slug(nav_blog)}"
+                                t-attf-class="#{navIsOffCanvas and 'nav-link' or 'dropdown-item'} #{(blog and nav_blog.id == blog.id) and 'active' or ''}"
```

Comment:
Either you indent like here OR like in line 29.
I do not enforce a way or another, but at least let's try to be consistent within the same file 😅